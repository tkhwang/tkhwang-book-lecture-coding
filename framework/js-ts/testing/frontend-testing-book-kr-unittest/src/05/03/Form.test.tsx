import { fireEvent, logRoles, render, screen } from "@testing-library/react";
import { Form } from "./Form";

/* 코드 5-3
test("이름을 표시한다", () => {
  render(<Form name="taro" />);
});
*/

/* 코드 5-4
test("이름을 표시한다", () => {
  render(<Form name="taro" />);
  console.log(screen.getByText("taro"));
});
*/

/* 코드 5-7
test("heading을 표시한다", () => {
  render(<Form name="taro" />);
  expect(screen.getByRole("heading"));
});
*/

/* 코드 5-53
test("Snapshot: 계정명인 'taro'가 표시됐는지 확인한다", () => {
  const { container } = render(<Form name="jiro" />);
  expect(container).toMatchSnapshot();
});
*/

test("이름을 표시한다", () => {
  render(<Form name="taro" />);
  expect(screen.getByText("taro")).toBeInTheDocument();
});

test("버튼을 표시한다", () => {
  render(<Form name="taro" />);
  expect(screen.getByRole("button")).toBeInTheDocument();
});

test("heading을 표시한다", () => {
  render(<Form name="taro" />);
  expect(screen.getByRole("heading")).toHaveTextContent("계정 정보");
});

test("버튼을 클릭하면 이벤트 핸들러가 실행된다", () => {
  const mockFn = jest.fn();
  render(<Form name="taro" onSubmit={mockFn} />);
  fireEvent.click(screen.getByRole("button"));
  expect(mockFn).toHaveBeenCalled();
});

test("Snapshot: 계정명인 'taro'가 표시된다", () => {
  const { container } = render(<Form name="taro" />);
  expect(container).toMatchSnapshot();
});

test("logRoles: 렌더링 결과로부터 역할과 접근 가능한 이름을 확인한다", () => {
  const { container } = render(<Form name="taro" />);
  logRoles(container);
});
