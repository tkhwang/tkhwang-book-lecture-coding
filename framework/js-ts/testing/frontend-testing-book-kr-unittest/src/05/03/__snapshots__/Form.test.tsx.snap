// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 계정명인 'taro'가 표시된다 1`] = `
<div>
  <form>
    <h2>
      계정 정보
    </h2>
    <p>
      taro
    </p>
    <div>
      <button>
        수정
      </button>
    </div>
  </form>
</div>
`;
